crates/ruff_linter/src/rules/fastapi/rules/fastapi_unused_path_parameter.rs

```diff
@@ -136,12 +136,26 @@ pub(crate) fn fastapi_unused_path_parameter(
     // Extract the path parameters from the route path.
     let path_params = PathParamIterator::new(path.to_str());
 
+    let dependables = keywordable_parameters(function_def)
+        .filter_map(|parameter_with_default| {
+            match Dependable::from_parameter(parameter_with_default, checker.semantic()) {
+                Dependable::None => None,
+                dependable => Some(dependable),
+            }
+        })
+        .collect::<Vec<_>>();
+
+    if dependables.iter().any(Dependable::is_unknown) {
+        return;
+    }
+
+    let dependables_parameter_names: Vec<_> = dependables
+        .into_iter()
+        .flat_map(Dependable::parameter_names)
+        .collect();
+
     // Extract the arguments from the function signature
-    let named_args: Vec<_> = function_def
-        .parameters
-        .args
-        .iter()
-        .chain(&function_def.parameters.kwonlyargs)
+    let named_args: Vec<_> = keywordable_parameters(function_def)
```

Comment:
I think we should just collect the dependencies directly into the `named_args` vector instead of building a second vector (and iterating over all parameters again).